src/huggingface_hub/_commit_api.py

```diff
@@ -449,9 +483,61 @@ def fetch_upload_modes(
     return upload_modes
 
 
+@validate_hf_hub_args
+def fetch_files_to_copy(
+    copies: Iterable[CommitOperationCopy],
+    repo_type: str,
+    repo_id: str,
+    token: Optional[str],
+    revision: str,
+    endpoint: Optional[str] = None,
+) -> Dict[Tuple[str, Optional[str]], "RepoFile"]:
+    """
+    Requests the Hub "preupload" endpoint to determine whether each input file
+    should be uploaded as a regular git blob or as git LFS blob.
+
+    Args:
+        copies (`Iterable` of :class:`CommitOperationCopy`):
+            Iterable of :class:`CommitOperationCopy` describing the files to
+            copy on the Hub.
+        repo_type (`str`):
+            Type of the repo to upload to: `"model"`, `"dataset"` or `"space"`.
+        repo_id (`str`):
+            A namespace (user or an organization) and a repo name separated
+            by a `/`.
+        token (`str`, *optional*):
+            An authentication token ( See https://huggingface.co/settings/tokens )
+        revision (`str`):
+            The git revision to upload the files to. Can be any valid git revision.
+
+    Returns: `Dict[Tuple[str, Optional[str]], RepoFile]]`
+        Key is the file path and revision of the file to copy, value is the repo file.
+
+    Raises:
+        [`~utils.HfHubHTTPError`]
+            If the Hub API returned an error.
+        [`ValueError`](https://docs.python.org/3/library/exceptions.html#ValueError)
+            If the Hub API response is improperly formatted.
+    """
+    from .hf_api import HfApi
+
+    hf_api = HfApi(endpoint=endpoint, token=token)
+    files_to_copy = {}
+    for src_revision, operations in groupby(copies, key=lambda op: op.src_revision):
+        operations = list(operations)  # type: ignore
+        paths = [op.src_path_in_repo for op in operations]
+        src_repo_files = hf_api.list_files_info(
+            repo_id=repo_id, paths=paths, revision=src_revision or revision, repo_type=repo_type
```

Comment:
```suggestion
            repo_id=repo_id, paths=paths, revision=src_revision, repo_type=repo_type
```

IMO we shouldn't default to the passed `revision` that the user is committing to. This would mean the a `CommitOperationCopy` is different depending on which branch we apply it. (Also it is not consistent with the `CommitOperationCopy.revision` docstring which says "default to main branch"). Better to assume that is `op.revision` is None then we copy from main branch.

Reply:
> This would mean the a CommitOperationCopy is different depending on which branch we apply it.

this would be consistent with git - the copies you do depend on your branch right ? :p 

> Also it is not consistent with the CommitOperationCopy.revision docstring which says "default to main branch"

oops my bad - I didn't mean that